### PR TITLE
Upgrade to .NET 7

### PR DESCRIPTION
**Pull request by xingsy97**
### Summary of the changes
 - Add .Net 7 support for some packages

Preparation for Client Invocation

### CODE REVIEW

File: run.ps1 [hunk 1]
```diff
@@ -124,10 +124,10 @@ function Get-KoreBuild {
             }
 			 
             # Hack to use Core version
-            Write-Host "!!! Hack to use .NET Core SDK 5.0.301"
```
Comment (JialinXin) [inline run.ps1:127]: update `run.sh` as well for Linux CI build.

File: src/Microsoft.Azure.SignalR/Microsoft.Azure.SignalR.csproj [hunk 2]
```diff
@@ -44,4 +44,8 @@
     <PackageReference Include="Microsoft.AspNetCore.Localization" Version="$(MicrosoftAspNetCoreLocalizationPackageVersion)" />
     <PackageReference Include="Microsoft.AspNetCore.SignalR" Version="$(MicrosoftAspNetCoreSignalRPackageVersion)" />
   </ItemGroup>
+	<ItemGroup Condition=" '$(TargetFramework)' == 'net7.0'">
```
Comment (JialinXin) [inline src/Microsoft.Azure.SignalR/Microsoft.Azure.SignalR.csproj:47]: use whitespace instead of tabs.

**Comment by xingsy97** [comment]
still a draft without need for draft

File: src/Microsoft.Azure.SignalR/Microsoft.Azure.SignalR.csproj (hunk 2 above)
Comment (xingsy97) [reply src/Microsoft.Azure.SignalR/Microsoft.Azure.SignalR.csproj]: fixed

File: run.ps1 (hunk 1 above)
Comment (xingsy97) [reply run.ps1:127]: fixed

Comment (JialinXin) [inline run.ps1:127]: Shall we upgrade to latest versions? See version information: https://github.com/dotnet/installer.

File: run.ps1 [hunk 3]
```diff
@@ -138,6 +138,36 @@ function Get-KoreBuild {
...
     }
 
+    # Hack $korebuildPath\modules\vstest\module.targets Line 147. This line executes `dotnet vstest ... --framework:...`. 
+    # When .NET SDK version >= 6.0, .NET CLI (`dotnet`) cannot handle parameter option `--framework` with `vstest` correctly.
+    # A easy solution is to replace `vstest` with `/test`. Another solution is to replace `--Framework` with `/Framework`
+    $hackTargetPath = "$korebuildPath\modules\vstest\module.targets"
+    (Get-Content $hackTargetPath -Raw) -Replace '<VSTestArgs Include="vstest" />', '<VSTestArgs Include="test" />' | Set-Content $hackTargetPath
+
+    # Forcibly Install .NET SDK 5.0.301
+    $textToAdd = {
```
Comment (JialinXin) [inline run.ps1:148]: Is this 5.0 SDK used by Tests projects? Shall we upgrade tests to net7.0 as well so we don't need to install this?

File: build/dependencies.props [hunk 4]
```diff
@@ -8,6 +8,7 @@
     <MicrosoftAspNetCoreHttpConnectionsClientPackage3_0Version>3.0.0</MicrosoftAspNetCoreHttpConnectionsClientPackage3_0Version>
     <MicrosoftAspNetCoreHttpConnectionsClientPackage3_1Version>3.1.9</MicrosoftAspNetCoreHttpConnectionsClientPackage3_1Version>
     <MicrosoftAspNetCoreHttpConnectionsClientPackage5_0Version>5.0.1</MicrosoftAspNetCoreHttpConnectionsClientPackage5_0Version>
+    <MicrosoftAspNetCoreHttpConnectionsClientPackage7_0Version>7.0.0-preview.7.22376.6</MicrosoftAspNetCoreHttpConnectionsClientPackage7_0Version>
```
Comment (xingsy97) [inline build/dependencies.props:11]: Althouth .NET SDK latest version is 7.0.100-preview.7.22377.5
the latest version of `Microsoft.AspNetCore.Http.Connections.Client` is 7.0.0-preview.7.22376.6 (refer to [Nuget.org](https://www.nuget.org/packages/Microsoft.AspNetCore.Http.Connections.Client/#versions-body-tab))

Comment (JialinXin) [reply build/dependencies.props:11]: Not a problem. We'll upgrade to net 7.0 release version before our final release. Make it up to date to easier find potential breaks in an early stage.

**Review by JialinXin — APPROVED** [review]

**Comment by xingsy97** [comment]
@JialinXin  I encountered [this error](https://github.com/Azure/azure-signalr/runs/8033391901?check_suite_focus=true#step:5:388) 
```
 Failed to negotiate protocol, waiting for response timed out after 90 seconds. This may occur due to machine slowness, please set environment variable VSTEST_CONNECTION_TIMEOUT to increase timeout.
  Results File: D:\a\azure-signalr\azure-signalr\artifacts\logs\UnitTests-net7.0-t000.trx
```
when executing `build.cmd`. The error is caused by vstest

In my local environment, I set this environment variable to `10000` by `$env:VSTEST_CONNECTION_TIMEOUT=10000` in powershell terminal and execute `build.cmd`. Then the test stuck here
```
  [xUnit.net 00:00:11.36]     Microsoft.Azure.SignalR.Common.Tests.Auth.AzureActiveDirectoryTests.TestAuthenticateAsync [SKIP]
  [xUnit.net 00:00:11.36]     Microsoft.Azure.SignalR.Common.Tests.Auth.AzureActiveDirectoryTests.TestGetAzureAdTokenAndAuthenticate [SKIP]
  [xUnit.net 00:00:11.36]     Microsoft.Azure.SignalR.Common.Tests.Auth.AzureActiveDirectoryTests.TestAcquireAccessToken [SKIP]
    Skipped Microsoft.Azure.SignalR.Common.Tests.Auth.AzureActiveDirectoryTests.TestAuthenticateAsync [1 ms]
    Skipped Microsoft.Azure.SignalR.Common.Tests.Auth.AzureActiveDirectoryTests.TestGetAzureAdTokenAndAuthenticate [1 ms]
    Skipped Microsoft.Azure.SignalR.Common.Tests.Auth.AzureActiveDirectoryTests.TestAcquireAccessToken [1 ms]
  [xUnit.net 00:00:20.43]     Microsoft.Azure.SignalR.Tests.TestEndpointServiceConnectionContainerTests.TestStatusPingChangesEndpointStatus [SKIP]
    Skipped Microsoft.Azure.SignalR.Tests.TestEndpointServiceConnectionContainerTests.TestStatusPingChangesEndpointStatus [1 ms]
```

**Pull request merged**